bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/OuterMapDeserializer.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.storage.json;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+/**
+ * Deserializes the internal data structure of the {@link JsonStorage})
+ *
+ * The contained entities remain json objects and won't me deserialized to their corresponding types at this point.
+ *
+ * @author Simon Kaufmann - initial contribution and API
+ *
+ */
+public class OuterMapDeserializer implements JsonDeserializer<Map<String, Object>> {
+
+    /**
+     *
+     * Finds out whether the given object is the outer JSON storage map or not.
+     *
+     * It must be
+     * <li>a Map of Maps
+     * <li>with 2 entries each
+     * <li>with {@link JsonStorage#CLASS} and {@link JsonStorage#VALUE} being their keys
+     *
+     * @param obj the object to be analyzed
+     * @return {@code true} if it is the outer storage map
+     */
+    private boolean isOuterMap(JsonObject obj) {
```

Comment:
I´m unsure about method ordering, do we want private methods grouped at the end of the class? or is there no ESH default?